ts_src/psetv2/bip371.ts

```diff
@@ -0,0 +1,89 @@
+import { TapLeafScript, TapScriptSig } from './interfaces';
+import { Input } from './input';
+import { tapLeafHash } from '../bip341';
+import { pubkeyPositionInScript } from './utils';
+
+export const toXOnly = (pubKey: Buffer) =>
+  pubKey.length === 32 ? pubKey : pubKey.slice(1, 33);
```

Comment:
would be best to have a switch to check the case for 32, 33, and then as failover default case throwing an error?

Reply:
This is taken from bitcoinJS, but I agree with you and will commit the change.